Library/Formula/riak.rb

```diff
@@ -29,4 +28,128 @@ def install
+  end
+
+  def maxfiles_plist; <<-EOS.undent
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+          <key>Label</key>
+            <string>limit.maxfiles</string>
+          <key>ProgramArguments</key>
+            <array>
+              <string>launchctl</string>
+              <string>limit</string>
+              <string>maxfiles</string>
+              <string>#{recommended_maxfiles}</string>
+              <string>#{recommended_maxfiles}</string>
+            </array>
+          <key>RunAtLoad</key>
+            <true/>
+          <key>ServiceIPC</key>
+            <false/>
+        </dict>
+      </plist>
+    EOS
+  end
+
+  def maxproc_plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple/DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+          <key>Label</key>
+            <string>limit.maxproc</string>
+          <key>ProgramArguments</key>
+            <array>
+              <string>launchctl</string>
+              <string>limit</string>
+              <string>maxproc</string>
+              <string>#{recommended_maxproc}</string>
+              <string>#{recommended_maxproc}</string>
+            </array>
+          <key>RunAtLoad</key>
+            <true/>
+          <key>ServiceIPC</key>
+            <false/>
+        </dict>
+      </plist>
+    EOS
+  end
+
+  def yosemite_maxfile_advice
+    puts "Please run \`sudo launchctl limit maxfiles #{recommended_maxfiles} #{recommended_maxfiles}\` or set the limits permanately with the following."
+    puts
+
+    puts "/Library/LaunchDaemons/limit.maxfiles.plist"
+    puts maxfiles_plist
+    puts
+  end
+
+  def yosemite_maxproc_advice
+    puts "Please run \`sudo launchctl limit maxproc #{recommended_maxproc} #{recommended_maxproc}\` or set the limits permanately with the following."
+    puts
+
+    puts "/Library/LaunchDaemons/limit.maxproc.plist"
+    puts maxproc_plist
+    puts
+  end
+
+  def mavericks_maxfile_advice
+    puts "limit maxfiles #{recommended_maxfiles} #{recommended_maxfiles}"
+  end
+
+  def mavericks_maxproc_advice
+    puts "limit maxproc #{recommended_maxproc} #{recommended_maxproc}"
+  end
+
+  def check_maxfiles
+    maxfiles = `launchctl limit maxfiles`
+    _, maxfiles_soft, maxfiles_hard = maxfiles.split
+
+    if maxfiles_soft.to_i < recommended_maxfiles || maxfiles_hard.to_i < recommended_maxfiles
+      puts
+      puts "Your OSX file limits are too low."
+
+      # How you change ulimits in OSX has changed with Yosemite
+      if MacOS.version == :yosemite
+        yosemite_maxfile_advice
+      else
+        puts "Edit or create /etc/launchd.conf with the following contents:"
+        mavericks_maxfile_advice  # mavericks and older actually
+      end
+    end
+  end
+
+  def post_install
+    check_maxfiles
+    check_maxproc
```

Comment:
I'll move it to the caveats.  Thanks.

Regarding the issue, it's from the [riak docs](http://docs.basho.com/riak/latest/ops/tuning/open-files-limit/).  Developers have to hunt this information later after getting warning messages on first startup.